django/db/backends/utils.py

```diff
@@ -53,6 +55,7 @@ def callproc(self, procname, params=None, kparams=None):
                 "Keyword parameters for callproc are not supported on this "
                 "database backend."
             )
+        self._warn_if_apps_not_ready()
```

Comment:
To avoid the cost of an extra function call for each query, can we inline the check into each of `callproc`, `execute`, and `executemany`? Like:

```python
if not apps.ready:
    self._warn_apps_not_ready()
```

Python function calls are relatively expensive, so if we can avoid them on the normal path we can be much surer there'll be no noticeable performance regression.

Reply:
We can also add a module level constant with the message, and remove `_warn_apps_not_ready()`, e.g.
```python
if not apps.ready:
    warnings.warn(APPS_NOT_READY_MSG, category=RuntimeWarning)
```

Reply:
Is there a particular reason for the module-level constant? I'd prefer a (private?) constant on the `CursorWrapper`, because nothing else in the module is using that message.